temba/flows/views.py

```diff
@@ -2149,11 +2166,43 @@ class RunTable(AllowOnlyActiveFlowMixin, OrgObjPermsMixin, SmartReadView):
 
         paginate_by = 50
 
+        @classmethod
+        def search_query(cls, query, base_queryset):
+            from .search.parser import FlowRunSearch
+
+            runs_search = FlowRunSearch(query=query, base_queryset=base_queryset)
+            return runs_search.search()
+
         def get_context_data(self, *args, **kwargs):
             context = super().get_context_data(*args, **kwargs)
             flow = self.get_object()
+
+            after = self.request.GET.get("after")
+            before = self.request.GET.get("before")
+            search_query = self.request.GET.get("q")
+            contact_query = self.request.GET.get("contact_query")
+
             runs = flow.runs.all()
 
+            if after:
+                after = str_to_datetime(after, tz=flow.org.timezone, dayfirst=flow.org.date_format == "D")
+                runs = runs.filter(created_on__gte=after)
+
+            if before:
+                before = str_to_datetime(before, tz=flow.org.timezone, dayfirst=flow.org.date_format == "D")
+                runs = runs.filter(created_on__lte=before)
```

Comment:
Per Rob recommendation, we should change from `created_on` to `modified_on` for the before/after filter